docs/build-the-builder.md

```diff
@@ -0,0 +1,186 @@
+
+```
+$ cp /usr/local/bin/bazel ~/kubevirt/hack/builder/bazel
+```
+
+## Build and publish the Bazel builder container
+
+### Set environment variables for building the builder -- all architectures, all registries
+
+Note that the default system architecture is amd64, so if you do not set ARCH as an environment variable, you will build a docker container with x86 object format binaries in it. 
+
+```
+export ARCH="<your-system-architecture>"
+export DOCKER_PREFIX="<your-registry-URL>/<your-kubevirt-namespace>" 
+export DOCKER_IMAGE="builder" 
+export VERSION="<your-docker-tag>"
+export KUBEVIRT_BUILDER_IMAGE="${DOCKER_PREFIX}/${DOCKER_IMAGE}/${VERSION}"
+```
+where <your-system-architecture> is either `amd64` or `arm64` or `s390x` or `ppc64le` for x86, ARM, s390x or Power systems, respectively.
+
+
+`make builder-build` invokes the build of the bazel builder container. 
+
+This tags your image with namespace you set up in your environment variables.
+
+Now you can
+
+```
+$ make builder-publish
+```
+
+which, among many other things, pushes the image up to the container registry of choice. Note that you will be pushing these changes to your test repository, in order to use this builder image to build kubevirt test operators in further steps. 
+
+### Update the rpms
+
+edit your local copy of this section of the file `hack/rpm-deps.sh`:
+
+```
+LIBVIRT_VERSION=${LIBVIRT_VERSION:-0:9.5.0-5.el9}
+QEMU_VERSION=${QEMU_VERSION:-17:8.0.0-13.el9}
+SEABIOS_VERSION=${SEABIOS_VERSION:-0:1.16.1-1.el9}
+EDK2_VERSION=${EDK2_VERSION:-0:20221207gitfff6d81270b5-9}
+LIBGUESTFS_VERSION=${LIBGUESTFS_VERSION:-1:1.48.4-4.el9}
+GUESTFSTOOLS_VERSION=${GUESTFSTOOLS_VERSION:-0:1.48.2-8.el9}
+PASST_VERSION=${PASST_VERSION:-0:0^20230818.g0af928e-4.el9}
+VIRTIOFSD_VERSION=${VIRTIOFSD_VERSION:-0:1.5.0-1.el9}
+SWTPM_VERSION=${SWTPM_VERSION:-0:0.8.0-1.el9}
+SINGLE_ARCH=${SINGLE_ARCH:-""}
+BASESYSTEM=${BASESYSTEM:-"centos-stream-release"}
+```
+
+Update the version numbers of the rpms in accordance with the latest versions observable in https://mirror.stream.centos.org/9-stream/AppStream/$ARCH/os/Packages/
+
+If you do not have a version that can be looked up in the CentOS rpm collection, the build of the bazel helper container for KubeVirt will fail. 
+
+Once you get it working, you can consider incorporating your local versions of the RPM dependencies in a PR, but careful to check for CVEs and other issues in the dependencies you specified. 
+
+### Obtain the `bazeldnf` utility
+
+If you are building the builder on Power, x86 or arm architecture you can get a pre-built binary of the `bazeldnf` utility from https://github.com/rmohr/bazeldnf/releases
+
+If you are building the builder container natively on s390x, you will need to compile it from source, which is also provided at https://github.com/rmohr/bazeldnf/releases
+
+
+Now run a script which you can name `./fix-rpm-deps.sh` to update these versions throughout the installation with bazeldnf:
```

Comment:
How about making this a proper script, as in `./hack/fix-rpm-deps-s490x.sh` and referencing it here? Also I'd suggest using `$1` as input for `arch` then.

Reply:
I'd like to put that in a separate PR, since it solves a different problem, and I think  pinning the dependency versions in the repo will require reviewing the dependencies for security issues.  This is just a suggestion for getting a test version of the builder and KubeVirt operator going for internal test.

Reply:
Agreed, this can be solved in another PR.